pygccxml/parser/scanner.py

```diff
@@ -186,6 +193,19 @@ def xml_generator_from_xml_file(self):
     def read(self):
         xml.sax.parse(self.xml_file, self)
 
+    def _handle_comment(self, declaration):
+        comm_decl = self.__declarations.get(declaration.comment)
+        if comm_decl:
+            with open(self.__files.get(comm_decl.location.file_name, "r")) as file:
```

Comment:
This should be optimized to only open and read a file once for a batch of symbols, ideally.

The simplest solution is to just cache the process file content either globally, or in this scanner object if that permits it.

Reply:
The file is now read into memory the first time and access from there each time it is utilized again.